test/identify/index.spec.ts

```diff
@@ -590,9 +597,15 @@ describe('Identify', () => {
 
       const identityServiceIdentifySpy = sinon.spy(libp2p.identifyService, 'identify')
       const identityServicePushSpy = sinon.spy(libp2p.identifyService, 'push')
-
+      const connectionPromise = pDefer()
+      libp2p.connectionManager.addEventListener('peer:connect', () => {
+        connectionPromise.resolve()
+      }, { once: true })
       const connection = await libp2p.dial(remoteAddr)
+
       expect(connection).to.exist()
+      // Wait for connection event to be emitted
+      await connectionPromise.promise
```

Comment:
Please use `pEvent` instead of `pDefer` here, it's more concise and better reflects the intention behind the code.